src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -167,8 +167,93 @@ message WatchResourcesUpdate {
 
 ////////////////////////////////////////////
 
+message ConsoleLogLine {
+    string text = 1;
+    optional bool is_error = 2;
+}
+
+// Initiates a subscription for the logs of a resource.
+message WatchResourceConsoleLogsRequest {
+    // Specifies the resource to watch logs from.
+    ResourceId resource_id = 1;
+}
+
+// A message received from the server when watching resource logs.
+// Contains potentially many lines to be appended to the log.
+message WatchResourceConsoleLogsUpdate {
+    repeated ConsoleLogLine log_lines = 1;
+}
+
+////////////////////////////////////////////
+
+message StructuredLog {
```

Comment:
Is the dashboard no longer going to host the OTLP server?

Reply:
If there's an existing service for this, I'd like to use it. Can you provide some more info?

Reply:
Telemetry data is pushed into the dashboard with OTLP. I thought the dashboard was going to continue to host the OTLP server and store its data in memory. Has that changed?

Reply:
Seems like there's different ideas here. @karolz-ms suggested that logs/telemetry might go over the same gRPC channel. I have been focussing mostly on the resource data flow for now, and will continue with that in the coming days. That'll give us time to discuss and work out the design here (and give me time to do a little more research).

Will the OTLP approach work in ACA to the best of your knowledge? It would be good to not implement the server side twice, once in the app host and again for ACA.

Reply:
> Seems like there's different ideas here. @karolz-ms suggested that logs/telemetry might go over the same gRPC channel.

What is the goal?

There are big implications to moving OTEL telemetry out of the dashboard. The service implementor would need to:
* Collect and store the OTEL data. They would probably need to implement an OTLP service like we've already done in the dashboard.
* Be responsible for storing the collected OTEL data in memory
* Implement the gRPC methods to so the dashboard can query the data

It wouldn't be difficult to move C# code from the dashboard project to the hosting project to fulfill this list. But a cloud provider would have a lot of work ahead of them to do this from scratch.

I wrote a doc about how OTEL works in Aspire: https://github.com/dotnet/aspire/blob/main/docs/open-telemetry-architecture.md

Reply:
> Seems like there's different ideas here. @karolz-ms suggested that logs/telemetry might go over the same gRPC channel. 

There must have been some miscommunication--I did not ever intend to suggest such thing ☺️

Reply:
We shouldn't touch the OTLP server part of the dashboard.

Reply:
@karolz-ms apologies, I misremembered. It was [this comment](https://github.com/dotnet/aspire/pull/1274#discussion_r1419902127) from @daniv-msft (not you) that prompted the addition of structure log support to the protocol.

Thanks all for levelling up my understanding. I've removed structured logs from the proto here and have a better idea of what we're aiming for. The console log support that @daniv-msft suggested remains.